### PR TITLE
fly 2.3.1 (new formula)

### PR DESCRIPTION
**Pull request by a1dutch**
- [Y] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [Y] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [Y] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [Y] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

---

This pull request, enables the concourse (https://concourse.ci) fly cli

### CODE REVIEW

File: Formula/fly.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Fly < Formula
+  desc "Concourse fly cli"
+  homepage "https://concourse.ci"
+  url "https://github.com/concourse/concourse/releases/download/v2.3.1/fly_darwin_amd64"
```
Comment (bfontaine) [inline Formula/fly.rb]: We need to build from source.

File: Formula/fly.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  sha256 "4940c41cddc94887a2a838e3c4259e5d6ff4448550d2a680621518c14a648039"
+
+  def install
+    bin.install "fly_darwin_amd64" => "fly"
+  end
+
+  test do
+    system "#{bin}/fly", "help"
```
Comment (bfontaine) [inline Formula/fly.rb]: Can’t we write a test that does something more substantial than e.g. `help` or `version`?

File: Formula/fly.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Fly < Formula
+  desc "Concourse fly cli"
```
Comment (bfontaine) [inline Formula/fly.rb]: `"Command line interface to Concourse"` might be more descriptive.

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/fly.rb (hunk 3 above)
Comment (a1dutch) [inline Formula/fly.rb]: 👍

File: Formula/fly.rb (hunk 1 above)
Comment (bfontaine) [inline Formula/fly.rb]: > it doesn't have a tagged release with the source

The last one is here: https://github.com/concourse/concourse/archive/v2.3.1.tar.gz

> and also uses direnv

`direnv` is available in Homebrew Core. You can add it as a dependency.

File: Formula/fly.rb (hunk 2 above)
Comment (bfontaine) [inline Formula/fly.rb]: I mean a test that actually does something; not one that just prints the current version or the help.

File: Formula/fly.rb (hunk 1 above)
Comment (a1dutch) [inline Formula/fly.rb]: The source for fly isn't in that archive, as its a git submodule.

**Comment by a1dutch** [comment]
@bfontaine thanks for your help/time but i will just close this pull request and open one for the caskroom at a later date.

File: Formula/fly.rb (hunk 1 above)
Comment (MikeMcQuaid) [inline Formula/fly.rb]: @a1dutch You can use a Git checkout with the Homebrew formula, too.

**Comment by a1dutch** [comment]
I have managed to change it to build from the source, but the submodules (including fly) are building from the head and not the released version, not sure how to fix this, any ideas?

Regarding the test, only thing we can use is `fly help` or `fly --version`

File: Formula/fly.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+class Fly < Formula
+  desc "Command line interface to Concourse"
+  homepage "https://concourse.ci"
+  url "https://github.com/concourse/concourse.git",
+    :tag => "v2.3.1",
+    :revision => "0fc7e2f6d46412ea70c8a9d81671c66d27276e44"
+
+  bottle :unneeded
```
Comment (MikeMcQuaid) [inline Formula/fly.rb]: We want a bottle for a `go` application.

File: Formula/fly.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  
+  depends_on "go"
```
Comment (MikeMcQuaid) [inline Formula/fly.rb]: `depends_on "go" => :build`

File: Formula/fly.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+  def install
+    cd "src/github.com/concourse/fly" do
```
Comment (MikeMcQuaid) [inline Formula/fly.rb]: `cd "src/github.com/concourse/fly"` and then skip the `do`, `end` and indentation.

File: Formula/fly.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+      ENV["GOPATH"] = buildpath
+      system "go", "build", "-o", "fly"
+      bin.install "fly"
+    end
+  end
+
+  test do
+    system "#{bin}/fly", "help"
```
Comment (MikeMcQuaid) [inline Formula/fly.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by bfontaine** [comment]
Ping re Mike’s comments above?

**Comment by woodruffw** [comment]
Closing due to inactivity.

**Pull request closed without merging**